Streak, Stats changes

parthanjaria opened the pull request:
Added streak and edited my stats UI.

parthanjaria committed:
Merge branch 'master' of https://github.com/green-freedom-pvt-ltd/share into Parth

parthanjaria committed:
Merge remote-tracking branch 'origin/master' into Parth

parthanjaria committed:
Cause sort logic changed.(to show not completed to the left and completed on the right)

added dialog_badge_layout

parthanjaria committed:
Issue solved of multiple times tracking showing, when 1st time user allows permission and location on.

parthanjaria committed:
added streak and goal functionality

parthanjaria committed:
streak goal setting in user details

parthanjaria committed:
my stats, added logic for daily,weekly and monthly.

Facing problem in naming the x-axis values when changing the values from daily<->weekly<->monthly

parthanjaria committed:
streak add, logic change.

parthanjaria committed:
my stats, daily/weekly/monthly : setting data and x-axis value

parthanjaria committed:
Added reminder in settings.

Logout crash solved

Added 3 charts in Profile fragment to see daily/weekly/monthly

parthanjaria committed:
Added Onboarding screens

parthanjaria committed:
Added/Edited Onboarding screens - goals,reminder(ADDED) and edited others

parthanjaria committed:
Bugs solved for onboarding/profile/reminder

parthanjaria committed:
Streak UI updated,
Added dialog in flow if changes not made,
Content changed,
bugs/suggestions done

parthanjaria committed:
Image added in share,
unsync runs shown(in history and leader board),
added pull to refresh in history page

ankit209 commented on app/src/main/java/com/sharesmile/share/core/MainActivity.java:
```diff
@@ -431,84 +454,124 @@ public void onClick(View v) {
             ActivityCompat.finishAffinity(this);
         } else {
-            super.onBackPressed();
+            FragmentManager fragmentManager = getSupportFragmentManager();
```
Made this code cleaner and a bit more readable:

```
@Override
    public void onBackPressed() {
        if (isDrawerOpened()) {
            mDrawerLayout.closeDrawer(Gravity.START);
            return;
        }
        hideKeyboard(null);
        if (getFragmentManager().getBackStackEntryCount() == 1) {
            ActivityCompat.finishAffinity(this);
            return;
        } else {
            FragmentManager fragmentManager = getSupportFragmentManager();
            if(fragmentManager.getBackStackEntryCount()>0) {
                Fragment fragment = getTopFragmentInBackStack();
                if (fragment instanceof StreakGoalFragment) {
                    StreakGoalFragment streakGoalFragment = (StreakGoalFragment) fragment;
                    if (streakGoalFragment.goals.get(streakGoalFragment.streakGoalAdapter.getPosition()).getId()
                            != MainApplication.getInstance().getUserDetails().getStreakGoalID()) {
                        displayStreakGoalNotSavedAlert();
                        return;
                    }
                }
            }
            super.onBackPressed();
        }
    }

    private void displayStreakGoalNotSavedAlert(){
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setMessage("Changes are not saved.");
        builder.setNegativeButton("DISCARD", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                getSupportFragmentManager().popBackStack();
            }
        });
        builder.setPositiveButton("SAVE", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                StreakGoalFragment fragment = (StreakGoalFragment) getTopFragmentInBackStack();
                fragment.saveGoal();
            }
        });
        builder.create().show();
    }

    private Fragment getTopFragmentInBackStack(){
        FragmentManager fragmentManager = getSupportFragmentManager();
        return fragmentManager.findFragmentByTag(fragmentManager.getBackStackEntryAt(fragmentManager.getBackStackEntryCount() - 1).getName());
    }
```

parthanjaria committed:
scroll bug solved in workout history

parthanjaria committed:
Increased padding for streak,
change textsize of menu item in save goal and profile edit,
text changed in set reminder

parthanjaria committed:
Issues solved,
Added events,
Added overlay guide to streak and stats

parthanjaria committed:
changed overlay on streak and stats,

new user profile screen changed,

streak icon sizes changed

parthanjaria committed:
Added dialog to edit height/weight n birthday, other issues solved.

parthanjaria committed:
URLS changed to node

parthanjaria committed:
crash issue solved by changeing imageview src to srcCompat

parthanjaria committed:
urls added of node

parthanjaria committed:
few crash solved from crashlytics

parthanjaria committed:
build 2.8.1

parthanjaria committed:
build 2.8.1.1

Onboarding logic changed(will get always after login), deleted images not used in drawable

parthanjaria committed:
build 2.8.1.2
usain bolt messages changed.

parthanjaria committed:
badge and acheived badge table added,

working on badge and acheived badge logic on homeScreenFragment

parthanjaria committed:
build 2.8.1.3,

crash issue solved, of user data not coming due to some reason, so checking if  user data present before going forward

parthanjaria committed:
Database Generated files

parthanjaria committed:
checking badges and added acheived badges

parthanjaria committed:
crash solved

parthanjaria committed:
crash solved

nehanm committed:
give change maker badge

nehanm committed:
give changemaker,streak,cause,marathon badges

nehanm committed:
show badge, sending data from service to fragment

nehanm committed:
show badges screens afted achieving

nehanm committed:
showing achieved badges in profile

nehanm committed:
crash solved in realRunFragment

nehanm committed:
version code changed

nehanm committed:
adding charity overview and achieved badges on profile screen

nehanm committed:
achievement screen with hall of fame.

nehanm committed:
Merge branch 'Parth_NodeAPI' into Parth_Badges

nehanm committed:
badge logic

parthanjaria committed:
badge click show badge details.

parthanjaria committed:
Events added,
Bugs solved,
Achived badge logic issue solved

parthanjaria committed:
New Badge Logic added, where bagde is linked to cause.

parthanjaria committed:
Sync Badges to server
Sync onboarding data to server
Get Charity overview data
Badges UI changes
Edit profile V2

parthanjaria committed:
Badges data
Issues and flows fixed

parthanjaria committed:
crash solved
Version change

parthanjaria committed:
Added CropImage in the flow.

parthanjaria committed:
Image Compression completed

parthanjaria committed:
Loading bar chart using AsyncTaskLoader
changing Rs. to  currency selected and kms to distance type selected

parthanjaria committed:
Added Star for category overview.

parthanjaria committed:
Streak_goal_id, remindertime and body_height_unit from server

parthanjaria committed:
Added Title and AchievedTitle table

Logic of title

API hit :
get badges and title

get category

Login: gmail clear user login saved data, so that always ask to user.

parthanjaria committed:
sync achievement badges and title to server.
change logic of login, sync all required data, then allow user in.

parthanjaria committed:
logic image change

parthanjaria committed:
Show title after achievements

Sync Achievement change logic

parthanjaria committed:
more polygon added

parthanjaria committed:
hack for cause name and in progress coming form server

parthanjaria committed:
crashes solved

Hevin1993 committed:
added tables to migration
crash solved
Version change

parthanjaria committed:
crashes solved

parthanjaria committed:
Fb crash solve

Charity overview twerk in logic

AchievedTitle issue solved

parthanjaria committed:
Badge and title, getting new description

Charity,oveview getting image

parthanjaria committed:
Streak issue solve

parthanjaria committed:
getting sponsor and partner from charity overview

parthanjaria committed:
Sync service changed.

parthanjaria committed:
UI changes

parthanjaria committed:
UI changes

Change in getting the streak in login logic

parthanjaria committed:
setting cause name error solved
2.8.3.0

parthanjaria committed:
string init to null

parthanjaria committed:
added colors to charity overview.
camera crash solved
other crashes solved

parthanjaria committed:
charityoverview design changes

parthanjaria committed:
sort charity overview causes

parthanjaria committed:
achievement badge design set
bugs solved

parthanjaria committed:
bugs solved

parthanjaria committed:
UI changes, Doble badges being generated issue solved

parthanjaria committed:
UI changes, Streak issue resolved, 2.8.3.7

parthanjaria committed:
charity Overview full page scroll, profile my impact/level max change, 2.8.3.8

parthanjaria committed:
Kotlin version changed from 1.1.51->1.2.30, impact league issues solved.

parthanjaria committed:
sdk version chaanged to 27, supportlibraryversion changed from 26.1.0 to 27.1.1, playservicesvrsion from 11.6.0 to 15.0.1

parthanjaria committed:
share crash solved

parthanjaria committed:
changed to picasso 2.71828 and to okhttp3, added uses-feature in manifest

parthanjaria committed:
change if condition to elseif

parthanjaria committed:
added param to save streak value of the user

parthanjaria committed:
formatting of code done

parthanjaria committed:
Merge branch 'Parth_Badges_location' into Parth_Badges

parthanjaria committed:
events added, load all runs on refresh history

parthanjaria committed:
changed logic of loading achieved badges.

parthanjaria committed:
changed vesionnumber. 2.8.4.0 - > beta

parthanjaria committed:
Merge branch 'Parth_Badges' into Parth

# Conflicts:
#	app/build.gradle
#	app/src/main/java/com/sharesmile/share/login/LoginImpl.java
#	app/src/main/java/com/sharesmile/share/utils/Utils.java
#	app/src/main/res/values/strings.xml

parthanjaria committed:
changes in onboarding after merge from parth_badges branch

parthanjaria committed:
upgrade compile sdk and target sdk to 27, change in startforeground()

parthanjaria committed:
upgrade activityrecognitionapi to activityrecognitionclient, also changed simpledateformat with locale to english

parthanjaria committed:
changed elapsedTime from mm:ss to hh:mm:ss(when hour is 0), achievement badge design change.

parthanjaria committed:
1 streak showing "day".
rupee currency added code

parthanjaria committed:
Merge branch 'Parth_Badges' into Parth

parthanjaria committed:
Notification vibrate issue resolved for Android O, Stopped workoutserviceretaineralarm, 2.8.4.2 - > beta

parthanjaria committed:
Merge branch 'Parth_Badges' into Parth

# Conflicts:
#	app/src/main/java/com/sharesmile/share/tracking/ui/RealRunFragment.java

parthanjaria committed:
kotlin_version : 1.2.30 -> 1.2.50, and gradle version 3.1.3 - >3.1.4

parthanjaria committed:
Add events

parthanjaria committed:
Duplicate badges issue solve.

parthanjaria committed:
deleted the achived badges/title,badges/title master from db after logout, setting the charityoverview after the badges are loaded from the server.

parthanjaria committed:
removed focus of audio during workout

parthanjaria committed:
Streak issue for logout and login.
beta -> 2.8.4.3

parthanjaria committed:
changed android:scaleType="centerCrop" to android:scaleType="fitXY"

parthanjaria committed:
image set to centercrop, and achivedbadges logic changed

parthanjaria committed:
Duplicate badge Issue solved. 2.8.4.6

parthanjaria committed:
after workout history refresh, refresh loader of stats.

parthanjaria committed:
added event

parthanjaria committed:
added event

parthanjaria committed:
allow backup false in manifest

parthanjaria committed:
set pickers valu in onboarding.

parthanjaria committed:
Grey badges and level progress change

parthanjaria committed:
v2.8.4.9

parthanjaria committed:
set achievement methods as syncronized in utils.

parthanjaria committed:
v2.8.5.1

parthanjaria committed:
added adaptive icons

parthanjaria committed:
few crashes solved of getting textview null when app in background.

parthanjaria committed:
Merge branch 'Parth_Badges' into Parth

# Conflicts:
#	app/src/main/java/com/sharesmile/share/onboarding/OnBoardingActivity.java
#	app/src/main/java/com/sharesmile/share/tracking/ui/RealRunFragment.java
